src/modules/export.js

```diff
@@ -256,6 +277,52 @@ export const closeExport = () => ({
   type: CLOSE
 });
 
+/**
+ * Update export fields
+ * @api public
+ */
+export const updateFields = (fields) => ({
+  type: UPDATE_FIELDS,
+  fields: fields
+});
+
+/**
+ * Select fields to be exported
+ * @api public
+ */
+export const changeModalProgressStatus = (status) => ({
+  type: CHANGE_MODAL_PROGRESS_STATUS,
+  status: status
+});
+
+export const sampleFields = () => {
+  return (dispatch, getState) => {
+    const {
+      ns,
+      exportData,
+      dataService: { dataService }
+    } = getState();
+
+    const spec = exportData.isFullCollection
+      ? { filter: {} }
+      : exportData.query;
+
+    dataService.find(ns, spec.filter, {limit: 1}, function(findErr, docs) {
+      if (findErr) {
+        return onError(findErr);
+      }
+
+      const fields = Object.keys(docs[0]).sort().reduce((obj, field) => {
```

Comment:
What if `docs[0]` has nested objects?

Reply:
i can't pass nested fields to the driver (if i understand `projection` option properly), so I am just chose to ignore nested objects and only do top level fields, which _can_ be passed it to the driver.

Reply:
`projection` can take dot notation. I've been using the handy [`flat.flatten(obj)`](http://npm.im/flat)